core/services/mongodb-remote/src/test/java/com/mongodb/stitch/core/services/mongodb/remote/sync/internal/NamespaceChangeStreamListenerUnitTests.kt

```diff
@@ -40,7 +40,7 @@ class NamespaceChangeStreamListenerUnitTests {
         // assert the stream does not open since we have no document ids
         ctx.isLoggedIn = true
         assertFalse(namespaceChangeStreamListener.openStream())
-        verify(nsConfigMock, times(1)).synchronizedDocumentIds
+        verify(nsConfigMock, times(3)).synchronizedDocumentIds
```

Comment:
[q] what changed to make this happen? this isn't a performance hit right?

Reply:
Hmm... it is a slight performance hit. I moved `getSynchronizedDocumentIds` to the top of `openStream` to pre-read them before taking the lock. But that may not be necessary anymore.